presto-main/src/main/java/io/prestosql/execution/CreateTableTask.java

```diff
@@ -96,16 +96,16 @@ public String explain(CreateTable statement, List<Expression> parameters)
             List<Expression> parameters,
             WarningCollector warningCollector)
     {
-        return internalExecute(statement, metadata, accessControl, stateMachine.getSession(), parameters);
+        return internalExecute(statement, metadata, accessControl, stateMachine.getSession(), parameters, warningCollector);
     }
 
     @VisibleForTesting
-    ListenableFuture<?> internalExecute(CreateTable statement, Metadata metadata, AccessControl accessControl, Session session, List<Expression> parameters)
+    ListenableFuture<?> internalExecute(CreateTable statement, Metadata metadata, AccessControl accessControl, Session session, List<Expression> parameters, WarningCollector warningCollector)
     {
         checkArgument(!statement.getElements().isEmpty(), "no columns for table");
 
         Map<NodeRef<Parameter>, Expression> parameterLookup = parameterExtractor(statement, parameters);
-        QualifiedObjectName tableName = createQualifiedObjectName(session, statement, statement.getName());
+        QualifiedObjectName tableName = metadata.redirectTable(session, createQualifiedObjectName(session, statement, statement.getName()), warningCollector);
```

Comment:
I have a general-level concern that the requirement to call `metadata.redirectTable` before calling a any other method on `metadata` can be error prone in practice, as it seems to easy not to call the `redirectTable` and overlook the missing call also in the review process. I am sure you have thought about this very deeply already... if you happen to have any notes about alternatives considered, i would be curious to learn more.

Reply:
This concern does make sense. I had a similar feeling when I was implementing this PR. However, the table redirection approach seems to be an appropriate one to address the hive-iceberg migration issue. A few other approaches that have been discussed:
 - Federation connector: Introduce a new connector which federates between Hive and Iceberg Connectors.
 - [Search path support](https://github.com/prestosql/presto/issues/4442#issuecomment-658454075)
cc @electrum

Reply:
Thanks.
But also if we settle up on redirect, we could try to model the API so that there is a help from type system to make sure redirection is considered. 
Like we have in functions -- function name, resolved function, etc.